VMAccess/vmaccess.py

```diff
@@ -397,29 +399,26 @@ def _fsck_check(hutil):
         hutil.do_exit(1, 'Check', 'error', '0', 'Check failed.')
 
 
-def _fsck_repair(hutil):
+def _fsck_repair(disk_name, hutil):
     # first unmount disks and loop devices lazy + forced
     try:
-        cmd_result = waagent.Run("umount -adfr")
+        cmd_result = waagent.Run("umount -f /%s" % disk_name)
         if cmd_result!=0:
             # Fail fast
-            raise Exception("Failed to unmount disks")
+            hutil.log("Failed to unmount disk")
+            raise Exception("Failed to unmount disk %s" % disk_name)
         else:
             # run repair
             retcode = waagent.Run("fsck -AR -y")
-        if retcode > 0:
-            raise Exception("Failed to repair disk")
-        else:
-            # remount all post repair and return
-            retcode = waagent.Run("mount -a")
+            hutil.log("Ran fsck")
         if retcode == 0:
-            return retcode
+            hutil.log(waagent.Run("mount"))
```

Comment:
@paulmey is right.
waagent.run is executed twice in line 415 and 416. I think, it's not expected.
And we should log some describable message here instead of retcode only.